source/s3_default_meta_request.c

```diff
@@ -216,6 +218,10 @@ static int s_s3_meta_request_default_prepare_request(
         aws_s3_meta_request_read_body(meta_request, &request->request_body);
     }
 
+    if (meta_request->should_compute_content_md5) {
+        aws_s3_message_util_add_content_md5_header(meta_request->allocator, &request->request_body, message);
```

Comment:
Is this being done on the calling thread? If so that’s likely to throttle the caller at the speed we can perform an md5 (slow) on a single core.

Reply:
This step does get done in the client bootstrap ELG right now, but we have a task (which we have done some work on) to make this step of the meta request (the prepare) optionally happen async/different ELG/thread.  tldr: this is the best place to do the md5.